task-impls/src/network.rs

```diff
@@ -107,9 +117,17 @@ impl<
                     }
                 },
             },
-            MessageKind::Data(_) => {
-                warn!("Got unexpected message type in network task!");
-                return;
+            MessageKind::Data(message) => {
+                match message {
+                    // ED Why do we need the view number in the transaction?
```

Comment:
Found this comment in this enum:
```
/// TODO rethink this when we start to send these messages
/// we only need the view number for broadcast
```
which is suggesting quite the opposite--keeping `view_number` and removing `transaction`. I guess we should rethink this. :) This is fine for now though.